bindings/java/src/junit/com/apple/foundationdb/tuple/TupleEncodingTest.java

```diff
@@ -0,0 +1,526 @@
+/*
+ * TupleEncodingTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb.tuple;
+
+import java.lang.reflect.Field;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import com.apple.foundationdb.FDB;
+import com.apple.foundationdb.FDBLibraryRule;
+import com.apple.foundationdb.subspace.Subspace;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests around packing, versionstamps, and assorted encoding-related stuff for tuples.
+ */
+@RunWith(Theories.class)
+public class TupleEncodingTest {
```

Comment:
I suspect that if we did everything that https://github.com/FoundationDB/fdb-record-layer/blob/main/fdb-extensions/src/test/java/com/apple/foundationdb/tuple/TupleTest.java does here we could retire that class and close https://github.com/FoundationDB/fdb-record-layer/issues/18.

Reply:
I opted to leave `TupleTest.java` there mainly because it has a test function involving an older API version, and a test involving the database itself (as opposed to purely the Tuple API).

Happy to delete those tests, but didn't want to do so unilaterally.

Reply:
✅